src/java/com/wix/rulesjvm/test_discovery/DiscoveredTestSuite.java

```diff
@@ -0,0 +1,212 @@
+package com.wix.rulesjvm.test_discovery;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.lang.reflect.Modifier;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.jar.JarEntry;
+import java.util.jar.JarInputStream;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@RunWith(PrefixSuffixTestDiscoveringSuite.class)
+class DiscoveredTestSuite {
+}
+
+class PrefixSuffixTestDiscoveringSuiteObject {
+  Class<?>[] discoverClasses() {
+    String[] archives = archivesPath().split(",");
+    Class<?>[] classes = Arrays.stream(archives).flatMap(this::discoverClassesIn).toArray(Class<?>[]::new);
+    if (classes.length == 0)
+      throw new IllegalStateException("Was not able to discover any classes " +
+              "for archives=" + Arrays.stream(archives).collect(Collectors.joining(",")) + ", " +
+              "prefixes=" + prefixes() + ", " +
+              "suffixes=" + suffixes());
+    return classes;
+  }
+
+  private Stream<Class<?>> discoverClassesIn(String archivePath) {
+    JarInputStream archive = archiveInputStream(archivePath);
```

Comment:
One thing that java has over scala is [try with resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html). Do you want to use it or add a TODO to move to it?

Reply:
Do you like this better?
```suggestion
    Class<?>[] classes = null;

    try (JarInputStream archive = archiveInputStream(archivePath)) {
      classes = discoverClasses(archive, prefixes(), suffixesWithClassSuffix());
    } catch (IOException e) {
      e.printStackTrace();
      throw new RuntimeException(e);
    }
    
    if (printDiscoveredClasses()) {
      System.out.println("Discovered classes:");
      Arrays.stream(classes).forEach(c -> System.out.println(c.getName()));
    }
    return Arrays.stream(classes);
```

seems like defensive code around `classes` nullity should be added regardless...